File: work/Main3.java
import java.util.*;

public class Main3 {
	public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		boolean[] yet = new boolean[200001];
		int count = 0;
		int now = 0;
		for(int i = 0; i < n; i++) {
			now =sc.nextInt();
			yet[now] = true;
			while(yet[count]){
				count++;
			}
			System.out.println(count);
		}
	}
}

File: work/main3.c
#include<stdio.h>
#include<stdlib.h>
#include<stdbool.h>

int main(void) {
	int n;
	scanf("%d", &n);
	bool yet[200001] = {};
	int count = 0;
	int now = 0;
	for (int i = 0; i < n; i++)
	{
		scanf("%d", &now);
		yet[now] = true;
		while (yet[count])
		{
			count++;
		}
		printf("%d\n", count);
	}
	return 0;
}